Panic in OverlayFsType::create when mounting overlayfs with non-existent lowerdir

### Describe the bug

The kernel panics when attempting to mount an overlay filesystem if one of the directories specified in the lowerdir option does not exist. The panic is caused by an unwrap() call on an error result (specifically ENOENT) during path parsing/resolution within the OverlayFsType::create function.

# https://github.com/asterinas/asterinas/blob/d9f4229e6c690c48f0a9b055665aadaefcf3d702/kernel/src/fs/overlayfs/fs.rs#L1188

### To Reproduce

1. Compile the program and run.

```
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <sys/mount.h>
#include <errno.h>
#include <sys/stat.h>

int main() {
    // Create directories for overlay filesystem setup
    system("mkdir -p /tmp/overlay_upper /tmp/overlay_work /tmp/overlay_mount");
    
    // Create a non-existent lower directory path to trigger the panic
    // This will cause FsPath::try_from() to fail or fs.lookup() to fail
    // leading to unwrap() panic in overlayfs create function
    const char* nonexistent_lower = "/tmp/nonexistent_lower_dir_12345";
    
    // Prepare mount options for overlay filesystem
    // The lowerdir points to a non-existent directory which should trigger the panic
    char mount_opts[1024];
    snprintf(mount_opts, sizeof(mount_opts), 
             "lowerdir=%s,upperdir=/tmp/overlay_upper,workdir=/tmp/overlay_work",
             nonexistent_lower);
    
    printf("Attempting to mount overlayfs with invalid lowerdir to trigger panic...\n");
    printf("Mount options: %s\n", mount_opts);
    
    // Attempt to mount overlay filesystem
    // This should trigger the panic in overlayfs create function at line 1189
    // when it tries to lookup the non-existent lower directory
    int result = mount("overlay", "/tmp/overlay_mount", "overlay", 0, mount_opts);
    
    if (result == 0) {
        printf("Mount succeeded (unexpected)\n");
        // Clean up
        umount("/tmp/overlay_mount");
    } else {
        printf("Mount failed with error: %s\n", strerror(errno));
    }
    
    // Clean up directories
    system("rm -rf /tmp/overlay_upper /tmp/overlay_work /tmp/overlay_mount");
    
    return 0;
}

```

### Environment

- Official docker asterinas/asterinas:0.17.0-20251217
- Asterinas version: main [Fix panic in `add_posix_timer` · asterinas/asterinas@d9f4229](https://github.com/asterinas/asterinas/commit/d9f4229e6c690c48f0a9b055665aadaefcf3d702)

### Logs

```
/etc/workflow_pocs # ./ex1579___home__yuchen__workspace__asterinas__kernel__src_
_fs__overlayfs__fs.rs_1189_27 
Attempting to mount overlayfs with invalid lowerdir to trigger panic...
Mount options: lowerdir=/tmp/nonexistent_lower_dir_12345,upperdir=/tmp/overlay_upper,workdir=/tmp/overlay_work
[    41.437] ERROR: Uncaught panic:
        called `Result::unwrap()` on an `Err` value: Error { errno: ENOENT, msg: Some("path resolution did not reach the final target") }
        at /root/asterinas/kernel/src/fs/overlayfs/fs.rs:1188
        on CPU 0 by thread Some(Thread { task: (Weak), data: Any { .. }, is_exited: false, cpu_affinity: AtomicIdSet { bits: [1], phantom: PhantomData<ostd::cpu::id::CpuId> }, sched_attr: SchedAttr { policy: SchedPolicyState { kind: AtomicSchedPolicyKind(2), policy: UnsafeCell { .. } }, last_cpu: AtomicCpuId(0), real_time: RealTimeAttr { prio: 99, time_slice: 31490100 }, fair: FairAttr { weight: 1024, pending_weight: 0, vruntime: 406959172 } } })
Printing stack trace:
   1: fn 0xffffffff889b1050 - pc 0xffffffff889b109b / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f2d0;

   2: fn 0xffffffff8804c4c0 - pc 0xffffffff8804ccbd / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f2f0;

   3: fn 0xffffffff8804eac0 - pc 0xffffffff8804eaca / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f6c0;

   4: fn 0xffffffff8804b000 - pc 0xffffffff8804b00a / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f6d0;

   5: fn 0xffffffff88ae1a00 - pc 0xffffffff88ae1a46 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f6e0;

   6: fn 0xffffffff88adecd0 - pc 0xffffffff88aded66 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f720;

   7: fn 0xffffffff88251890 - pc 0xffffffff882519cf / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f7a0;

   8: fn 0xffffffff88056c10 - pc 0xffffffff88056c2c / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f870;

   9: fn 0xffffffff880fade0 - pc 0xffffffff880fae7e / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f8a0;

  10: fn 0xffffffff880526f0 - pc 0xffffffff88052749 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f930;

  11: fn 0xffffffff88055290 - pc 0xffffffff880552f0 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f990;

  12: fn 0xffffffff8819b320 - pc 0xffffffff8819b4eb / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6f9f0;

  13: fn 0xffffffff881af190 - pc 0xffffffff881af1bd / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6faf0;

  14: fn 0xffffffff881afb90 - pc 0xffffffff881afbf8 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6fb20;

  15: fn 0xffffffff881ac920 - pc 0xffffffff881ac950 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6fb90;

  16: fn 0xffffffff8819f130 - pc 0xffffffff8819f176 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6fbc0;

  17: fn 0xffffffff880551d0 - pc 0xffffffff88055283 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6fc10;

  18: fn 0xffffffff8825c8d0 - pc 0xffffffff8825cf32 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd6fc90;

  19: fn 0xffffffff88479700 - pc 0xffffffff88479b6a / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd70410;

  20: fn 0xffffffff88478230 - pc 0xffffffff88478493 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd708f0;

  21: fn 0xffffffff8847a0e0 - pc 0xffffffff8847aa3f / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd70b20;

  22: fn 0xffffffff88687980 - pc 0xffffffff886a9fdf / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd710f0;

  23: fn 0xffffffff88686f50 - pc 0xffffffff88686fea / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd844f0;

  24: fn 0xffffffff88296ad0 - pc 0xffffffff88297381 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd84650;

  25: fn 0xffffffff88297540 - pc 0xffffffff8829759e / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd84b40;

  26: fn 0xffffffff882ba290 - pc 0xffffffff882ba2be / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd84cf0;

  27: fn 0xffffffff882b8df0 - pc 0xffffffff882b8e36 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd84d20;

  28: fn 0xffffffff883a4660 - pc 0xffffffff883a468c / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd84d90;

  29: fn 0xffffffff883a1700 - pc 0xffffffff883a173f / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd84dd0;

  30: fn 0xffffffff88297500 - pc 0xffffffff88297527 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd84f40;

  31: fn 0xffffffff8809c1b0 - pc 0xffffffff8809c1cd / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd84f80;

  32: fn 0xffffffff889a1180 - pc 0xffffffff889a11a5 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd84fa0;

  33: fn 0xffffffff88978240 - pc 0xffffffff889782c4 / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd84fd0;

  34: fn 0xffffffff88971cb1 - pc 0xffffffff88971cba / registers:

     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd85000;


     rax 0xffffdfffffd6f1e8; rdx 0xffffffff889a1e22; rcx  0x10000020100762a; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bcf880; rbp                0x0; rsp 0xffffdfffffd85008;

OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) ostd::panic::print_stack_trace
          at /root/asterinas/ostd/src/panic.rs:113
(  2) aster_kernel::thread::oops::panic_handler
          at /root/asterinas/kernel/src/thread/oops.rs:130
(  3) __ostd_panic_handler
          at ??:?
(  4) __rustc::rust_begin_unwind
          at 86a8wkuc8764m1mx2x4getn5f:?
(  5) core::panicking::panic_fmt
          at ??:?
(  6) core::result::unwrap_failed
          at ??:?
(  7) <core::result::Result<aster_kernel::fs::path::Path, aster_kernel::error::Error>>::unwrap
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:1232
(  8) core::iter::adapters::map::map_fold::<&&str, aster_kernel::fs::path::Path, (), <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}, core::iter::traits::iterator::Iterator::for_each::call<aster_kernel::fs::path::Path, <alloc::vec::Vec<aster_kernel::fs::path::Path>>::extend_trusted<core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}>>::{closure#0}>::{closure#0}>::{closure#0}
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/adapters/map.rs:88
(  9) <core::slice::iter::Iter<&str> as core::iter::traits::iterator::Iterator>::fold::<(), core::iter::adapters::map::map_fold<&&str, aster_kernel::fs::path::Path, (), <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}, core::iter::traits::iterator::Iterator::for_each::call<aster_kernel::fs::path::Path, <alloc::vec::Vec<aster_kernel::fs::path::Path>>::extend_trusted<core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}>>::{closure#0}>::{closure#0}>::{closure#0}>
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:283
( 10) <core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}> as core::iter::traits::iterator::Iterator>::fold::<(), core::iter::traits::iterator::Iterator::for_each::call<aster_kernel::fs::path::Path, <alloc::vec::Vec<aster_kernel::fs::path::Path>>::extend_trusted<core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}>>::{closure#0}>::{closure#0}>
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/adapters/map.rs:129
( 11) <core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}> as core::iter::traits::iterator::Iterator>::for_each::<<alloc::vec::Vec<aster_kernel::fs::path::Path>>::extend_trusted<core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}>>::{closure#0}>
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs:829
( 12) <alloc::vec::Vec<aster_kernel::fs::path::Path>>::extend_trusted::<core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}>>
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:3876
( 13) <alloc::vec::Vec<aster_kernel::fs::path::Path> as alloc::vec::spec_extend::SpecExtend<aster_kernel::fs::path::Path, core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}>>>::spec_extend
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/spec_extend.rs:28
( 14) <alloc::vec::Vec<aster_kernel::fs::path::Path> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<aster_kernel::fs::path::Path, core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}>>>::from_iter
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/spec_from_iter_nested.rs:60
( 15) <alloc::vec::Vec<aster_kernel::fs::path::Path> as alloc::vec::spec_from_iter::SpecFromIter<aster_kernel::fs::path::Path, core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}>>>::from_iter
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/spec_from_iter.rs:33
( 16) <alloc::vec::Vec<aster_kernel::fs::path::Path> as core::iter::traits::collect::FromIterator<aster_kernel::fs::path::Path>>::from_iter::<core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}>>
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:3724
( 17) <core::iter::adapters::map::Map<core::slice::iter::Iter<&str>, <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create::{closure#0}> as core::iter::traits::iterator::Iterator>::collect::<alloc::vec::Vec<aster_kernel::fs::path::Path>>
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs:2028
( 18) <aster_kernel::fs::overlayfs::fs::OverlayFsType as aster_kernel::fs::registry::FsType>::create
          at /root/asterinas/kernel/src/fs/overlayfs/fs.rs:1189
( 19) aster_kernel::syscall::mount::get_fs
          at /root/asterinas/kernel/src/syscall/mount.rs:255
( 20) aster_kernel::syscall::mount::do_new_mount
          at /root/asterinas/kernel/src/syscall/mount.rs:202
( 21) aster_kernel::syscall::mount::sys_mount
          at /root/asterinas/kernel/src/syscall/mount.rs:65
( 22) aster_kernel::syscall::arch::syscall_dispatch
          at /root/asterinas/kernel/src/syscall/mod.rs:244
( 23) aster_kernel::syscall::handle_syscall
          at /root/asterinas/kernel/src/syscall/mod.rs:374
( 24) aster_kernel::thread::task::create_new_user_task::{closure#0}
          at /root/asterinas/kernel/src/thread/task.rs:95
( 25) aster_kernel::thread::task::create_new_user_task::{closure#1}
          at /root/asterinas/kernel/src/thread/task.rs:126
( 26) unwinding::panicking::catch_unwind::do_call::<aster_kernel::thread::task::create_new_user_task::{closure#1}, ()>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panicking.rs:56
( 27) unwinding::panicking::catch_unwind::<unwinding::panic::RustPanic, (), aster_kernel::thread::task::create_new_user_task::{closure#1}>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panicking.rs:42
( 28) unwinding::panic::catch_unwind::<(), aster_kernel::thread::task::create_new_user_task::{closure#1}>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panic.rs:87
( 29) aster_kernel::thread::oops::catch_panics_as_oops::<aster_kernel::thread::task::create_new_user_task::{closure#1}, ()>
          at /root/asterinas/kernel/src/thread/oops.rs:55
( 30) aster_kernel::thread::task::create_new_user_task::{closure#2}
          at /root/asterinas/kernel/src/thread/task.rs:131
( 31) <aster_kernel::thread::task::create_new_user_task::{closure#2} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
( 32) <alloc::boxed::Box<dyn core::ops::function::FnOnce<(), Output = ()> + core::marker::Send> as core::ops::function::FnOnce<()>>::call_once
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2206
( 33) kernel_task_entry
          at /root/asterinas/ostd/src/task/mod.rs:196
( 34) kernel_task_entry_wrapper
          at ??:?
make: *** [Makefile:309: run_kernel] Error 1
```

--- a/kernel/src/fs/overlayfs/fs.rs
+++ b/kernel/src/fs/overlayfs/fs.rs
@@ -1184,12 +1184,8 @@ impl FsType for OverlayFsType {
         let upper = path_resolver.lookup(&FsPath::try_from(upper)?)?;
         let lower = lower
             .iter()
-            .map(|&lower| {
-                path_resolver
-                    .lookup(&FsPath::try_from(lower).unwrap())
-                    .unwrap()
-            })
-            .collect();
+            .map(|&lower| path_resolver.lookup(&FsPath::try_from(lower)?))
+            .collect::<Result<Vec<_>>>()?;
         let work = path_resolver.lookup(&FsPath::try_from(work)?)?;
 
         OverlayFs::new(upper, lower, work).map(|fs| fs as _)

